README.md

```diff
@@ -153,36 +168,87 @@ class Entity(Base):
 
 Use [FastAPI dependency injection] to get a session as a parameter of a path operation
 function.
-SQLAlchemy session is committed before response is returned or rollbacked if any
+
+The SQLAlchemy session is committed before the response is returned or rollbacked if any
 exception occurred:
 
 ```python
+from fastapi import APIRouter
+from fastapi_sqla import AsyncSession, Session
+
+router = APIRouter()
+
+
+@router.get("/example")
+def example(session: Session):
+    return session.execute("SELECT now()").scalar()
+
+
+@router.get("/async_example")
+async def async_example(session: AsyncSession):
+    return await session.scalar("SELECT now()")
+```
+
+In order to get a session configured with a custom key:
+
+```python
+from typing import Annotated
+
 from fastapi import APIRouter, Depends
-from fastapi_sqla import Session, AsyncSession
+from fastapi_sqla import (
+    AsyncSessionDependency,
+    SessionDependency,
+    SqlaAsyncSession,
+    SqlaSession,
+)
 
 router = APIRouter()
 
 
+# Preferred
+
+read_only_session_dep = SessionDependency(key="read_only")
+ReadOnlySession = Annotated[SqlaSession, Depends(read_only_session_dep)]
+
+async_read_only_session_dep = AsyncSessionDependency(key="read_only")
+AsyncReadOnlySession = Annotated[
+    SqlaAsyncSession, Depends(async_read_only_session_dep)
+]
```

Comment:
I'd get back to how it was documented without separating the dep from annotation. Seems cleaner to me. And sorry for suggesting it had to be overridden in tests... YAGNI